### ISSUE TITLE
Mean values for continuous variables may become unordered in table

### ISSUE THREAD

**Issue by tompollard** [tompollard/tableone#3]
There may be an issue where mean values are incorrectly reported in the output table (describe table seems fine), due to incorrect sorting.

**Comment by alistairewj** [comment]
Reproducing example:

```python
n = 20
data_groups = pd.DataFrame(index=range(n))

data_groups['group'] = 'group1'
data_groups.loc[ 2:6, 'group'] = 'group2'
data_groups.loc[ 6:12, 'group'] = 'group3'
data_groups.loc[12: n, 'group'] = 'group4'

data_groups['age'] = range(n)
data_groups['weight'] = [x+100 for x in range(n)]
```

`data_groups` looks like:

index | group | age | weight
--- | --- | --- | ---
0 | group1 | 0 | 100
1 | group1 | 1 | 101
2 | group2 | 2 | 102
3 | group2 | 3 | 103
4 | group2 | 4 | 104
5 | group2 | 5 | 105
6 | group3 | 6 | 106
7 | group3 | 7 | 107
8 | group3 | 8 | 108
9 | group3 | 9 | 109
10 | group3 | 10 | 110
11 | group3 | 11 | 111
12 | group4 | 12 | 112
13 | group4 | 13 | 113
14 | group4 | 14 | 114
15 | group4 | 15 | 115
16 | group4 | 16 | 116
17 | group4 | 17 | 117
18 | group4 | 18 | 118
19 | group4 | 19 | 119


Using `groupby` to calculate the mean:

```python
data_groups.groupby('group').mean()

	age	weight
group		
group1	0.5	100.5
group2	3.5	103.5
group3	8.5	108.5
group4	15.5	115.5
```

TableOne call showing the ordering issue:

```python
continuous = ['age','weight']
categorical = []
strata_col = 'group'
TableOne(data_groups, continuous = continuous, categorical = categorical, strata_col = strata_col)

Stratified by group
                     group1         group2         group3         group4
-------------------  -------------  -------------  -------------  -------------
n                    2              4              6              8
age (mean (std))     15.50 (2.45)   0.50 (0.71)    8.50 (1.87)    3.50 (1.29)
weight (mean (std))  115.50 (2.45)  100.50 (0.71)  108.50 (1.87)  103.50 (1.29
```

Looks like..

* group1 is in group2
* group2 is in group4
* group3 is correct
* group4 is in group1

Interestingly the sample sizes are all correct!

**Comment by tompollard** [comment]
@alistairewj this is fixed in the current version (v0.2.4). The column names were being sorted, but the values were not. I'll do some refactoring to populate a dataframe instead of list of lists at some point, which will make things cleaner.
